### ISSUE TITLE
Getting very high top-1 accuracy on cifar100 which is not normal.

### ISSUE THREAD

**Issue by sabujlaskar** [chenyaofo/image-classification-codebase#3]
Hi @chenyaofo, Thanks for the project. I tried to evaluate with pre-trained models on the cifar100 dataset. But with resnet20, the top1 accuracy is around 92%. But as far as I know, the top-1 accuracy of cifar100 is in the range of 70%-75%. Below is the conf/cifar10 file.
""""
max_epochs: 200

log_interval: 10

use_amp: false # if true, it will train in automatic mixed precision mode

only_evaluate: false # if true, it will only evalute the model on the validation set and exit

auto_resume: true # if true, it will automatically load the checkpoint in the output directory and continue to train

sync_batchnorm: false # if true, it will convert all the batchnorm layers into torch.nn.SyncBatchNorm

accmulated_steps: 1

memory_format: channels_last

data {
    type_: cifar10

    is_vit: false

    image_size: 32
    num_classes: 10

    root: data/cifar10

    mean: [0.4914, 0.4822, 0.4465]
    std: [0.2023, 0.1994, 0.2010]

    batch_size: 256
    num_workers: 4
}

model {
    type_: PyTorchHub
    force_reload: false
    repo: chenyaofo/pytorch-cifar-models
    name: cifar10_resnet20
    pretrained: false
    load_from: null
}

optimizer {
    type_: CustomSGD
    lr: 0.01
    momentum: 0.9
    dampening: 0
    weight_decay: 5e-4
    nesterov: true
    basic_bs: 256
}

scheduler {
    type_: CosineAnnealingLR
    T_max: ${max_epochs}
    eta_min: 0
}

criterion {
    type_: CrossEntropyLoss
}
""""
conf/cifar100 file is same as repo. 

Below is the evaluation result:

2021-11-18 20:22:46,840 INFO    : EVAL, epoch=0000, iter=00010/00196, fetch data time cost=17.14ms, fps=235 images/s, loss=0.2921, top1-acc=93.20%, top5-acc=99.41%, remaining_time=0:03:22.968222, arrival_time=2021-11-18 20:26:09.808
329, cost_time=0:00:10.912267
2021-11-18 20:22:47,421 INFO    : EVAL, epoch=0000, iter=00020/00196, fetch data time cost=0.76ms, fps=4409 images/s, loss=0.2857, top1-acc=92.87%, top5-acc=99.41%, remaining_time=0:01:41.146496, arrival_time=2021-11-18 20:24:28.568
253, cost_time=0:00:11.493917
2021-11-18 20:22:47,978 INFO    : EVAL, epoch=0000, iter=00030/00196, fetch data time cost=0.38ms, fps=4605 images/s, loss=0.2794, top1-acc=93.10%, top5-acc=99.53%, remaining_time=0:01:06.672406, arrival_time=2021-11-18 20:23:54.650
453, cost_time=0:00:12.049231
2021-11-18 20:22:48,566 INFO    : EVAL, epoch=0000, iter=00040/00196, fetch data time cost=1.95ms, fps=4356 images/s, loss=0.2849, top1-acc=92.87%, top5-acc=99.53%, remaining_time=0:00:49.286952, arrival_time=2021-11-18 20:23:37.852
492, cost_time=0:00:12.637700
2021-11-18 20:22:49,122 INFO    : EVAL, epoch=0000, iter=00050/00196, fetch data time cost=0.38ms, fps=4611 images/s, loss=0.2827, top1-acc=92.85%, top5-acc=99.52%, remaining_time=0:00:38.529254, arrival_time=2021-11-18 20:23:27.652
059, cost_time=0:00:13.194965
2021-11-18 20:22:49,706 INFO    : EVAL, epoch=0000, iter=00060/00196, fetch data time cost=0.37ms, fps=4390 images/s, loss=0.2841, top1-acc=92.81%, top5-acc=99.49%, remaining_time=0:00:31.231448, arrival_time=2021-11-18 20:23:20.937
841, cost_time=0:00:13.778553
2021-11-18 20:22:50,278 INFO    : EVAL, epoch=0000, iter=00070/00196, fetch data time cost=0.46ms, fps=4478 images/s, loss=0.2862, top1-acc=92.75%, top5-acc=99.47%, remaining_time=0:00:25.830756, arrival_time=2021-11-18 20:23:16.109
040, cost_time=0:00:14.350444
2021-11-18 20:22:50,835 INFO    : EVAL, epoch=0000, iter=00080/00196, fetch data time cost=0.36ms, fps=4597 images/s, loss=0.2856, top1-acc=92.80%, top5-acc=99.46%, remaining_time=0:00:21.616136, arrival_time=2021-11-18 20:23:12.451
673, cost_time=0:00:14.907697
2021-11-18 20:22:51,391 INFO    : EVAL, epoch=0000, iter=00090/00196, fetch data time cost=0.37ms, fps=4609 images/s, loss=0.2893, top1-acc=92.63%, top5-acc=99.44%, remaining_time=0:00:18.213132, arrival_time=2021-11-18 20:23:09.604
944, cost_time=0:00:15.463972
2021-11-18 20:22:51,963 INFO    : EVAL, epoch=0000, iter=00100/00196, fetch data time cost=0.85ms, fps=4478 images/s, loss=0.2900, top1-acc=92.61%, top5-acc=99.45%, remaining_time=0:00:15.394464, arrival_time=2021-11-18 20:23:07.358
167, cost_time=0:00:16.035863
2021-11-18 20:22:52,538 INFO    : EVAL, epoch=0000, iter=00110/00196, fetch data time cost=0.64ms, fps=4456 images/s, loss=0.2886, top1-acc=92.67%, top5-acc=99.45%, remaining_time=0:00:12.986516, arrival_time=2021-11-18 20:23:05.525
038, cost_time=0:00:16.610682
2021-11-18 20:22:53,104 INFO    : EVAL, epoch=0000, iter=00120/00196, fetch data time cost=0.39ms, fps=4526 images/s, loss=0.2880, top1-acc=92.65%, top5-acc=99.48%, remaining_time=0:00:10.878564, arrival_time=2021-11-18 20:23:03.983
120, cost_time=0:00:17.176716
2021-11-18 20:22:53,680 INFO    : EVAL, epoch=0000, iter=00130/00196, fetch data time cost=0.39ms, fps=4451 images/s, loss=0.2878, top1-acc=92.64%, top5-acc=99.48%, remaining_time=0:00:09.012828, arrival_time=2021-11-18 20:23:02.693
180, cost_time=0:00:17.752512
2021-11-18 20:22:54,237 INFO    : EVAL, epoch=0000, iter=00140/00196, fetch data time cost=0.37ms, fps=4598 images/s, loss=0.2874, top1-acc=92.65%, top5-acc=99.49%, remaining_time=0:00:07.323904, arrival_time=2021-11-18 20:23:01.561
508, cost_time=0:00:18.309764
2021-11-18 20:22:54,825 INFO    : EVAL, epoch=0000, iter=00150/00196, fetch data time cost=0.52ms, fps=4362 images/s, loss=0.2881, top1-acc=92.60%, top5-acc=99.50%, remaining_time=0:00:05.795172, arrival_time=2021-11-18 20:23:00.620
281, cost_time=0:00:18.897269
2021-11-18 20:22:55,380 INFO    : EVAL, epoch=0000, iter=00160/00196, fetch data time cost=0.36ms, fps=4616 images/s, loss=0.2871, top1-acc=92.60%, top5-acc=99.51%, remaining_time=0:00:04.376844, arrival_time=2021-11-18 20:22:59.757
255, cost_time=0:00:19.452571
2021-11-18 20:22:55,987 INFO    : EVAL, epoch=0000, iter=00170/00196, fetch data time cost=0.75ms, fps=4218 images/s, loss=0.2868, top1-acc=92.59%, top5-acc=99.52%, remaining_time=0:00:03.067948, arrival_time=2021-11-18 20:22:59.055
381, cost_time=0:00:20.059593
2021-11-18 20:22:56,544 INFO    : EVAL, epoch=0000, iter=00180/00196, fetch data time cost=0.37ms, fps=4597 images/s, loss=0.2870, top1-acc=92.58%, top5-acc=99.51%, remaining_time=0:00:01.832608, arrival_time=2021-11-18 20:22:58.377
293, cost_time=0:00:20.616845
2021-11-18 20:22:57,115 INFO    : EVAL, epoch=0000, iter=00190/00196, fetch data time cost=0.40ms, fps=4485 images/s, loss=0.2864, top1-acc=92.59%, top5-acc=99.51%, remaining_time=0:00:00.669090, arrival_time=2021-11-18 20:22:57.784
691, cost_time=0:00:21.187761
2021-11-18 20:22:57,421 INFO    : EVAL, epoch=0000, iter=00196/00196, fetch data time cost=0.37ms, fps=4506 images/s, loss=0.2871, top1-acc=92.55%, top5-acc=99.51%, remaining_time=0:00:00, arrival_time=2021-11-18 20:22:57.421065, co
st_time=0:00:21.493225

I need to use this in my project. Any probable reason for this issue will be very helpful for me. Thank you.

**Comment by chenyaofo** [comment]
Sorry for this bug. I have fixed it in the latest version.

Note that the top-1 acc of resnet20 are 92.60% (on CIFAR-10) and 68.83% (on CIFAR-100). You can find top-1/5 acc of more models on my another repo https://github.com/chenyaofo/pytorch-cifar-models

Hope this can help you!
